modules/ml/src/main/java/org/apache/ignite/ml/sql/SQLFunctions.java

```diff
@@ -17,18 +17,29 @@
 
 package org.apache.ignite.ml.sql;
 
-import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
 import org.apache.ignite.cache.query.annotations.QuerySqlFunction;
 import org.apache.ignite.ml.inference.IgniteModelStorageUtil;
 import org.apache.ignite.ml.inference.Model;
 import org.apache.ignite.ml.math.primitives.vector.Vector;
 import org.apache.ignite.ml.math.primitives.vector.VectorUtils;
+import org.apache.ignite.ml.util.LRUCache;
 
 /**
  * SQL functions that should be defined and passed into cache configuration to extend list of functions available
  * in SQL interface.
  */
 public class SQLFunctions {
+    /** Default LRU model cache size. */
+    private static final int LRU_CACHE_SIZE = 10;
+
+    /** Default LRU model cache. */
+    // TODO: IGNITE-11163: Add hart beat tracker to DistributedInfModel.
+    private static final Map<String, Model<Vector, Double>> cache = Collections.synchronizedMap(
```

Comment:
what about models that was changed after you locally save older model?

Reply:
Added thread that clears the cache.